File: com.systemincloud.ext.vip.examples.tasks.javatask/src/test/java/tasks/data/DataImage.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.ext.vip.modeler.api.javatask.data.Image;

@JavaTaskInfo
public class DataImage extends JavaTask {

    private static int N = 100;
    
    @InputPortInfo(name = "In", dataType = Image.class)
    public InputPort in;
    @OutputPortInfo(name = "Out", dataType = Image.class)
    public OutputPort out;

    @Override
    public void execute(int grp) {
        Image img = in.getData(Image.class);
        int[] inValues = img.getValues();
        int[] outValues = new int[img.getNumberOfElements()];
        for(int i = 0; i < outValues.length; i++) {
            int inRGB = inValues[i];
            int r = (inRGB >> 16 & 0xff) + N;
            int g = (inRGB >> 8 & 0xff) + N;
            int b = (inRGB & 0xff) + N;
            if(r > 255) r = 255; if(r < 0) r = 0;
            if(g > 255) g = 255; if(g < 0) g = 0;
            if(b > 255) b = 255; if(b < 0) b = 0;
            outValues[i] = (r << 16) | (g << 8) | b;
        }
        out.putData(new Image(outValues, img.getH(), img.getW()));
    }

}

File: com.systemincloud.ext.vip.examples.tasks.pythontask/src/test/py/tasks/data/DataImage.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask_vip.data.Image import Image

@PythonTaskInfo
class DataImage(PythonTask):

    def __init_ports__(self):
        self.in_ = InputPort(name="in", data_type=Image)
        self.out = OutputPort(name="out", data_type=Image)
     
    def execute(self, grp):
        img = self.in_.get_data(Image)
        r = img.r
        g = img.g
        b = img.b
        self.out.put_data(Image.from_RGB(img.dims, r, g, b))
        
